telluric/georaster.py

```diff
@@ -1606,8 +1606,15 @@ def get_window(self, window, bands=None,
             with self._raster_opener(self._filename) as raster:  # type: rasterio.io.DatasetReader
                 array = raster.read(bands, **read_params)
             affine = affine or self._calculate_new_affine(window, out_shape[2], out_shape[1])
-
             raster = self.copy_with(image=array, affine=affine)
+
```

Comment:
I think you can do this check before reading the raster and then to do the read only if there is an intersection